opus/application/static_media/js/opus.js

```diff
@@ -724,11 +734,91 @@ var opus = {
+            // Windows: Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.10136
+            // Mac: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/76.0.3800.0 Safari/537.36 Edg/76.0.167.0
+            matchObj = userAgent.match(/Edge\/(\d+.\d+)/) || userAgent.match(/Edg\/(\d+.\d+)/);
+            browserName = "edge";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("OPR") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36 OPR/60.0.3255.95
+            matchObj = userAgent.match(/OPR\/(\d+.\d+)/);
+            browserName = "opera";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("Chrome") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36
+            matchObj = userAgent.match(/Chrome\/(\d+.\d+)/);
+            browserName = "chrome";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("Version") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/605.1.15
+            // (KHTML, like Gecko) Version/12.1 Safari/605.1.15
+            matchObj = userAgent.match(/Version\/(\d+.\d+)/);
+            browserName = "safari";
+            browserVersion = matchObj[1];
+        }
+
+        if (opus.browserSupport[browserName] === undefined) {
+            $("#op-browser-version-msg").modal("show");
+        } else {
+            if (parseFloat(browserVersion) < opus.browserSupport[browserName]) {
+                $("#op-browser-version-msg").modal("show");
+            }
+        }
+    },
 
+    checkBrowserSize: function() {
+        /**
+         * Check if browser width is less than 1280px. If so, display a
+         * modal to inform the user to resize the browser width.
+         */
+        if ($(window).width() < opus.browserSupport.width) {
+            $("#op-browser-size-msg").modal("show");
+        } else {
+            $("#op-browser-size-msg").modal("hide");
+        }
+    },
+
+    checkCookies: function() {
+        /**
+         * Check widgets cookie to determine if the user is a first time
+         * visitor. If so, we display a guide page.
+         * Note: we will call __help/splash.html api in the future to
+         * display the guide page. For now, we just show a modal.
+         */
+        if ($.cookie("widgets") === undefined) {
+            $("#op-guide").modal("show");
+        }
+    }
 }; // end opus namespace
 
 $(document).ready(function() {
+    opus.checkBrowserVersion();
+    opus.checkCookies();
```

Comment:
This is probably the place you need to above if checkBrowserVersion returns saying the browser is bad. Don't do anything else, including trying to initialize OPUS. Just give up.